symengine/expand.cpp

```diff
@@ -297,6 +298,26 @@ class ExpandVisitor : public BaseVisitor<ExpandVisitor>
             _coef_dict_add_term(multiply, r);
             return;
         }
+        if (is_a<Integer>(*self.get_exp()) && is_a<URatPoly>(*_base)) {
```

Comment:
Can you remove this? Polynomials are not still integrated into `add`, `mul` and `pow`, so let's deal with later. (We shouldn't have added the lines above as well.)